### PR TITLE
Allow plugins to use WKS header strings

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

File: doc/developer-guide/api/functions/TSMimeHdrFieldValueStringGet.en.rst [hunk 1]
```diff
@@ -65,6 +65,9 @@ preferred.
 value, and populated :arg:`value_len_ptr` with the length of the
 value in bytes. The returned header value is not NUL-terminated.
 
+In addition to all the predefined constants for Well-Kown header strings, you can
```
Comment (bneradt) [inline doc/developer-guide/api/functions/TSMimeHdrFieldValueStringGet.en.rst:68]: Kown -> Known

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks pretty straightforward.

I think I missed some discussion about this, but just to make sure I understand this PR. Are we adding `TSMimeHdrStringToWKS` because `TSMimeHdrFieldFind` is significantly more performant with a wks?

**Comment by SolidWallOfCode** [comment]
If we step back a bit, why can't `TS_MIME_FIELD_CACHE_CONTROL` be the WKS string? Would this then before the WKS that don't have an exported symbol?

**Comment by zwoop** [comment]
> Looks pretty straightforward.
> 
> I think I missed some discussion about this, but just to make sure I understand this PR. Are we adding `TSMimeHdrStringToWKS` because `TSMimeHdrFieldFind` is significantly more performant with a wks?

Yes, significantly faster (no strncmp's).

**Comment by zwoop** [comment]
@SolidWallOfCode Not sure I understand. TS_MIME_FIELD_CACHE_CONTROL is the WKS for "Cache-Control", MIME_FIELD_CACHE_CONTROL is set using the exact API that I'm exposing here to plugins.The problem is that there's no way to go from "Cache-Control" to TS_MIME_FIELD_CACHE_CONTROL, and that's what I'm exposing (making public).

Yes, will update the docs with the nullptr return value.

**Comment by zwoop** [comment]
Updated the docs, will email the mailing list now with the API proposal.

**Comment by ezelkow1** [comment]
[approve ci centos]

**Comment by zwoop** [comment]
@bneradt I fixed the typo, can you review again please?

**Review by bneradt — APPROVED** [review]

**Pull request merged**